app/src/main/java/com/duckduckgo/app/httpsupgrade/api/HttpsUpgradeListDownloader.kt

```diff
@@ -38,21 +51,47 @@ class HttpsUpgradeListDownloader @Inject constructor(
             val call = service.https()
             val response = call.execute()
 
-            if (response.isCached && httpsUpgradeDao.count() != 0) {
+            if (response.isCached && dbWriteStatusStore.hasWriteCompleted() && httpsUpgradeDao.count() != 0) {
                 Timber.d("HTTPS data already cached and stored")
                 return@fromAction
             }
 
             if (response.isSuccessful) {
                 Timber.d("Updating HTTPS upgrade list from server")
-                val domains = response.body()!!.toTypedArray()
-                database.runInTransaction {
+
+                val domains = response.body() ?: throw IllegalStateException("Failed to obtain HTTPS upgrade list")
+
+                val startTime = System.currentTimeMillis()
+
+                modifyHttpsDatabase {
                     httpsUpgradeDao.deleteAll()
-                    httpsUpgradeDao.insertAll(*domains)
+                    Timber.v("Took ${System.currentTimeMillis() - startTime}ms to delete existing records")
+
+                    val chunks = domains.chunked(chunkSize)
+                    Timber.i("Received ${domains.size} HTTPS domains; chunking by $chunkSize into ${chunks.size} separate DB transactions")
+
+                    chunks.forEach {
+                        database.runInTransaction {
+                            httpsUpgradeDao.insertAll(it)
```

Comment:
Is `it` really a special sygil here?

Reply:
`it` here is implicitly available because `chunks.forEach` takes a lambda with a single parameter. 

so `it` == `an individual chunk` in that scope